asmcomp/cmmgen_state.ml

```diff
@@ -76,6 +76,9 @@ let set_structured_constants l =
     )
     l
 
+let add_structured_constant sym cst =
+  Hashtbl.add state.structured_constants sym cst
```

Comment:
Since you mention there could be duplicate calls to this, it might be better to use Hashtbl.replace to avoid accumulating redundant bindings.

Reply:
Thanks for the suggestion. I've pushed a commit with this change.